logic/dapp.js

```diff
@@ -101,12 +101,11 @@ DApp.prototype.verify = function(transaction, sender, cb, tx) {
 	}
 
 	let foundCategory = false;
-	for (const i of Object.keys(dappCategories)) {
-		if (dappCategories[i] === transaction.asset.dapp.category) {
+	Object.keys(dappCategories).forEach(key => {
+		if (dappCategories[key] === transaction.asset.dapp.category) {
 			foundCategory = true;
-			break;
 		}
-	}
+	});
```

Comment:
Prefer using `find()` as it returns the first element in an array that pass a test and breaks the loop.

```
const foundCategory = !!Object.values(dappCategories).find(value => value === transaction.asset.dapp.category);
```

Reply:
We need to iterate over every category to check if all of them are valid.

Reply:
The original code does `break` when a category is found

Reply:
The original code does `break` when a category is found